File: 03-recursion_onlyevendigits-Java/recursion_onlyevendigits.java
// # Without using iteration and without using strings, 
// # write the recursive function onlyEvenDigits(L), 
// # that takes a array L of non-negative integers 
// # (you may assume that), and returns a new array of 
// # the same numbers only without their odd digits 
// # (if that leaves no digits, then replace the number with 0). 
// # So: onlyEvenDigits([43, 23265, 17, 58344]) returns [4, 226, 0, 844]. 
// # Also the function returns the empty array if the original array is empty. 
// # Remember to not use strings. You may not use loops/iteration in this problem.

class recursion_onlyevendigits {
	public int[] fun_recursion_onlyevendigits(int[] l){
		int[] arr = {};
		if(l.length == 0){
			return arr;
		}
		return arr;
	}

	public boolean odd_even(int n){
		if(n % 2 == 0){
			return true;
		}else{
			return false;
		}
	}

	public static void main(String[] args) {
		
	}
}

File: 03-recursion_onlyevendigits-Python/recursion_onlyevendigits.py
# Without using iteration and without using strings, 
# write the recursive function onlyEvenDigits(L), 
# that takes a list L of non-negative integers 
# (you may assume that), and returns a new list of 
# the same numbers only without their odd digits 
# (if that leaves no digits, then replace the number with 0). 
# So: onlyEvenDigits([43, 23265, 17, 58344]) returns [4, 226, 0, 844]. 
# Also the function returns the empty list if the original list is empty. 
# Remember to not use strings. You may not use loops/iteration in this problem.

def fun_recursion_onlyevendigits(l): 
	return recursion(l, [], 0)

def recursion(l, ar, i):
	if i == len(l):
		return ar
	else:
		num = l[i]
		r = []
		for i in range(num):
			rem = num % 10
			num = num // 10
			if rem % 2 == 0:
				r.append(rem)
		r = list(map(str, r[: : -1]))
		s = ''.join(r)
		if int(s) % 2 == 0:
			ar.append(int(s))
		i = i + 1
		return recursion(l, ar, i)